diff --git a/Gemfile b/Gemfile
index 0eb308a..dbb9995 100644
--- a/Gemfile
+++ b/Gemfile
@@ -9,6 +9,7 @@ gem 'rails', '~> 6.0.3', '>= 6.0.3.3'
 gem 'pg', '>= 0.18', '< 2.0'
 # Use Puma as the app server
 gem 'puma', '~> 4.1'
+gem 'mustache'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 # gem 'jbuilder', '~> 2.7'
 # Use Redis adapter to run Action Cable in production
diff --git a/Gemfile.lock b/Gemfile.lock
index 23077b2..c6e6a54 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -90,6 +90,7 @@ GEM
     mini_portile2 (2.4.0)
     minitest (5.14.2)
     msgpack (1.3.3)
+    mustache (1.1.1)
     nio4r (2.5.4)
     nokogiri (1.10.10)
       mini_portile2 (~> 2.4.0)
@@ -164,6 +165,7 @@ DEPENDENCIES
   dotenv-rails
   jwt (~> 2.2)
   listen (~> 3.2)
+  mustache
   pg (>= 0.18, < 2.0)
   puma (~> 4.1)
   rack-cors
diff --git a/app/models/campaign.rb b/app/models/campaign.rb
index ed245a4..12981e1 100644
--- a/app/models/campaign.rb
+++ b/app/models/campaign.rb
@@ -55,10 +55,11 @@ class Campaign < ApplicationRecord
 
   def send_to_segment
     fromAddress = SendGrid::Email.new(email: from)
-    body = SendGrid::Content.new(type: "text/html", value: content)
     sg = SendGrid::API.new(api_key: ENV["SENDGRID_API_KEY"])
 
     segment.contacts.each { |contact|
+      value = Mustache.render(content, name: contact.name)
+      body = SendGrid::Content.new(type: "text/html", value: value)
       campaign_contact = campaign_contacts.create(contact: contact)
       to = SendGrid::Email.new(email: contact.email)
       mail = SendGrid::Mail.new(fromAddress, subject, to, body)
diff --git a/client/src/Campaign/EditForm.js b/client/src/Campaign/EditForm.js
index a682158..4f4ac63 100644
--- a/client/src/Campaign/EditForm.js
+++ b/client/src/Campaign/EditForm.js
@@ -20,6 +20,8 @@ class EditForm extends React.Component {
 
   render() {
     const editorConfiguration = {
+      allowedContent: true,
+      height: 700,
       toolbar: [
         { name: 'document', items: ['Source'] },
         { name: 'clipboard', items: ['Cut', 'Copy', 'Paste', 'PasteText', 'Undo', 'Redo'] },
@@ -46,10 +48,10 @@ class EditForm extends React.Component {
         <h4>Campaign: {this.props.campaign.name}</h4>
         <h4>Subject: {this.props.campaign.subject}</h4>
         <h4>Segment: {this.props.campaign.segment.name}</h4>
-        <h4>Content</h4>
+        <small>*Use <i>{"{{name}}"}</i> markup to personalize email per contact</small>
         <Form onSubmit={this.submitHandler}>
           <Row>
-            <Col xs="9">
+            <Col>
               <FormGroup>
                 <CKEditor
                   data={this.props.campaign.content}
@@ -62,8 +64,12 @@ class EditForm extends React.Component {
                 />
               </FormGroup>
             </Col>
+          </Row>
+          <Row>
             <Col xs="3">
               <Button color="primary" className="redirect-btn">Save</Button>
+            </Col>
+            <Col xs="3">
               <Button color="secondary" className="redirect-btn" href={`/campaigns/${this.props.campaign.id}/preview`}>Go back to preview</Button>
             </Col>
           </Row>
diff --git a/public/templates/index_1.html b/public/templates/index_1.html
deleted file mode 100644
index 3a9c7bc..0000000
--- a/public/templates/index_1.html
+++ /dev/null
@@ -1 +0,0 @@
-<h1>Template 1</h1>
diff --git a/public/templates/index_2.html b/public/templates/index_2.html
deleted file mode 100644
index 5511873..0000000
--- a/public/templates/index_2.html
+++ /dev/null
@@ -1,2 +0,0 @@
-<h1>Template 2</h1>
-<div>hi</div>
diff --git a/public/templates/template_1.html b/public/templates/template_1.html
index adad9fa..4fdf2b2 100644
--- a/public/templates/template_1.html
+++ b/public/templates/template_1.html
@@ -2,7 +2,7 @@
   <br /><br /><br />
   <table
     style="background: #ffffff;"
-    width="840"
+    width="712"
     align="center"
     cellpadding="20"
   >
@@ -79,7 +79,7 @@
     style="font-size: 12px; color: #888888; font-family: Arial;"
   >
     All rights reserved. Copyright &copy; 2020. Your address here<br />
-    Don't like these emails? <a style="color: #888888;" href="#">Unsubsribe</a>
+    Don't like these emails? <a style="color: #888888;" href="#">Unsubscribe</a>
     <br />
     <br />
     Powered by Newsletter
diff --git a/public/templates/template_2.html b/public/templates/template_2.html
index c8b6f5a..3936785 100644
--- a/public/templates/template_2.html
+++ b/public/templates/template_2.html
@@ -2,7 +2,7 @@
   <br /><br /><br />
   <table
     style="background: #ffffff;"
-    width="840"
+    width="712"
     align="center"
     cellpadding="20"
   >
@@ -22,7 +22,7 @@
       <td style="font-family: Arial;">
         <img
           src="https://via.placeholder.com/800x450"
-          width="800"
+          width="672"
           height="450"
           alt=""
         />
@@ -78,7 +78,7 @@
     style="font-size: 12px; color: #888888; font-family: Arial;"
   >
     All rights reserved. Copyright &copy; 2020. Your address here<br />
-    Don't like these emails? <a style="color: #888888;" href="#">Unsubsribe</a>
+    Don't like these emails? <a style="color: #888888;" href="#">Unsubscribe</a>
     <br />
     <br />
     Powered by Newsletter
